File: HelloShoesBE/src/main/java/lk/ijse/HelloShoesBE/entity/Sale.java
package lk.ijse.HelloShoesBE.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import java.sql.Timestamp;
import java.util.HashSet;
import java.util.Set;

@AllArgsConstructor
@NoArgsConstructor
@Data
@Table(name = "sale")
@Entity
public class Sale implements SuperEntity{
    @Id
    private String orderCode;
    private String customerName;
    private double totalPrice;
    private Timestamp date;
    private String payMethod;
    private double addedPoints;
    private String cashierName;
    @ManyToOne
    @JoinColumn(name = "userCode")
    private User user;
    @OneToMany(mappedBy = "sale", cascade = CascadeType.ALL, orphanRemoval = true)
    private Set<SaleInventories> saleInventories = new HashSet<>();
    @ManyToOne
    @JoinColumn(name = "customerCode")
    private Customer customer;
}

File: HelloShoesFE/model/Sale.js
export class Sale {
    constructor(orderCode, customerName, totalPrice, date, payMethod, addedPoints, cashierName, userCode, customerCode, saleInventories = []) {
        this.orderCode = orderCode;
        this.customerName = customerName;
        this.totalPrice = totalPrice;
        this.date = date;
        this.payMethod = payMethod;
        this.addedPoints = addedPoints;
        this.cashierName = cashierName;
        this.userCode = userCode;
        this.customerCode = customerCode;
        this.saleInventories = saleInventories;
    }
}